cmf-cli/Builders/ProcessCommand.cs

```diff
@@ -48,41 +53,49 @@ public Task Exec()
 
                     command = exePath ?? command;
                 }
-                Log.Debug($"Executing '{command} {String.Join(' ', step.Args ?? Array.Empty<string>())}'");
-                ProcessStartInfo ps = new();
-                ps.FileName = command;
-                ps.WorkingDirectory = step.WorkingDirectory != null ? step.WorkingDirectory.FullName : this.WorkingDirectory.FullName;
-                ps.Arguments = String.Join(' ', step.Args);
-                ps.UseShellExecute = false;
-                ps.RedirectStandardOutput = true;
-                ps.RedirectStandardError = true;
 
-                if(step.EnvironmentVariables.HasAny())
+                if (Condition())
                 {
-                    foreach (var envVar in step.EnvironmentVariables)
+                    Log.Debug($"Executing '{command} {String.Join(' ', step.Args ?? Array.Empty<string>())}'");
+                    IProcessStartInfoCLI ps = ExecutionContext.ServiceProvider.GetService<IProcessStartInfoCLI>();

```

Comment:
not sure if we should remove the using statement, there are points where you can leave the execution path without disposing. maybe this should be handled in the ProcessStartInfoCLI implementation, though.